File: app/src/main/java/dev/ragnarok/fenrir/api/interfaces/IOtherApi.java
package dev.ragnarok.fenrir.api.interfaces;

import java.util.Map;

import dev.ragnarok.fenrir.util.Optional;
import io.reactivex.rxjava3.core.Single;

public interface IOtherApi {
    Single<Optional<String>> rawRequest(String method, Map<String, String> postParams);
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/api/interfaces/IOtherApi.kt
package dev.ragnarok.fenrir.api.interfaces

import dev.ragnarok.fenrir.util.Optional
import io.reactivex.rxjava3.core.Single
import okhttp3.ResponseBody

interface IOtherApi {
    fun rawRequest(method: String, postParams: Map<String, String>): Single<Optional<ResponseBody>>
}
